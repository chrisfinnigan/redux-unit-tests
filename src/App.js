import React, { Component } from "react";
import { connect } from "react-redux";
import UserForm from "./UserForm/UserForm";

import "./App.css";

export class App extends Component {
  buttonClickHandler = event => {};

  render() {
    return (
      <div className="App">
        <header className="">Header</header>
        <main>
          <UserForm />
          <pre>{JSON.stringify(this.props)}</pre>
        </main>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state
});

const mapDispatchToProps = dispatch => ({});

export default connect(mapStateToProps, mapDispatchToProps)(App);
