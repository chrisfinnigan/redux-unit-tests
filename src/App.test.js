import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { App } from "./App";
import configureStore from "./store";

describe("App", () => {
  it("should render in debug mode", () => {
    const tree = renderer
      .create(
        <Provider store={configureStore()}>
          <App debug />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
