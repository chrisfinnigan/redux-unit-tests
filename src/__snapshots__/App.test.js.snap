// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render in debug mode 1`] = `
<div
  className="App"
>
  <header
    className=""
  >
    Header
  </header>
  <main>
    <form>
      <label>
        First Name
        <input
          id="first_name"
          name="first_name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Last Name
        <input
          name="last_name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Age
        <input
          name="age"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </form>
    <pre>
      {"debug":true}
    </pre>
  </main>
</div>
`;
